spec/models/reservation_spec.rb

```diff
@@ -87,705 +73,830 @@
+        expect(res.human_status).to eq(string)
+      end
     end
+    it_behaves_like 'returns the proper string', 'starts today',
+                    :valid_reservation, start_date: Time.zone.today
+    it_behaves_like 'returns the proper string', 'reserved',
+                    :valid_reservation, start_date: Time.zone.today + 1.day
+    it_behaves_like 'returns the proper string', 'due today',
+                    :checked_out_reservation, due_date: Time.zone.today
+    it_behaves_like 'returns the proper string', 'checked_out',
+                    :checked_out_reservation, due_date: Time.zone.today + 1.day
+    it_behaves_like 'returns the proper string', 'returned overdue',
+                    :overdue_returned_reservation
+    it_behaves_like 'returns the proper string', 'overdue', :overdue_reservation
+    it_behaves_like 'returns the proper string', 'missed', :missed_reservation
+    it_behaves_like 'returns the proper string', 'returned',
+                    :checked_in_reservation
+    it_behaves_like 'returns the proper string', 'requested', :request
+    it_behaves_like 'returns the proper string', 'denied', :request,
+                    status: 'denied'
+  end
 
-    it { is_expected.not_to be_valid }
-    it 'should not save' do
-      expect(reservation.save).to be_falsey
-      expect(Reservation.all.size).to eq(1)
+  describe '#end_date' do
+    context 'if checked in' do
+      it 'returns the checkin date' do
+        res = FactoryGirl.build_stubbed(:checked_in_reservation)
+        expect(res.end_date).to eq(res.checked_in)
+      end
+      it 'does not care if overdue' do
+        res = FactoryGirl.build_stubbed(:overdue_returned_reservation)
+        expect(res.end_date).to eq(res.checked_in)
+      end
     end
-    it 'cannot be updated' do
-      reservation.start_date = Time.zone.today + 1.day
-      expect(reservation.save).to be_falsey
+    it 'returns today if actively overdue' do
+      res = FactoryGirl.build_stubbed(:overdue_reservation)
+      expect(res.end_date).to eq(Time.zone.today)
     end
-    it 'fails appropriate validations' do
-      expect(reservation.available).not_to eq([])
-      expect(reservation).not_to be_valid
+    it 'returns due date for request' do
+      res = FactoryGirl.build_stubbed(:request)
+      expect(res.end_date).to eq(res.due_date)
     end
-
-    it 'passes other custom validations' do
-      expect(reservation.start_date_before_due_date).to be_nil
-      expect(reservation.not_empty).to be_nil
-      expect(reservation.not_in_past).to be_nil
+    it 'returns due date for reserved' do
+      res = FactoryGirl.build_stubbed(:valid_reservation)
+      expect(res.end_date).to eq(res.due_date)
     end
   end
 
-  context 'with equipment item/model matching problems' do
-    subject(:reservation) do
-      r = FactoryGirl.build(:valid_reservation)
-      r.equipment_item = FactoryGirl.create(:equipment_item)
-      r
+  describe '#duration' do
```

Comment:
Just an aside, but duration should probably use `end_date` instead of `due_date`. I'll open up a new issue.